### PR TITLE
New version: Mozilla.Firefox.Nightly version 108.2211.1408.0

### LINKED ISSUE #87309: [Package Issue]: Remove Mozilla.Firefox.Nightly yamls
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

No applicable update found.

### Brief description of your issue

Firefox Nightly is a package that is updated twice a day.
It is not appropriate to add Mozilla.Firefox.Nightly to winget because we cannot handle such an update frequency, and in fact has not been updated for a long time.

### Steps to reproduce

winget show Mozilla.Firefox.Nightly

### Actual behavior

Show 97.0a1. on a particular release date.

### Expected behavior

Show the latest version on the latest date released.

### Environment

```shell
[winget --info]
Windows パッケージ マネージャー v1.3.2691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.755
システム アーキテクチャ: X64
パッケージ: Microsoft.DesktopAppInstaller v1.18.2691.0
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by russellbanks**
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

* Resolves #87309

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/87337)

### CODE REVIEW

**Review by jedieaston — DISMISSED** [review]

**Review by warshanks — DISMISSED** [review]

**Comment by Trenly** [comment]
You may need the package family name

**Comment by russellbanks** [comment]
> You may need the package family name

Will add after the weekend

**Review by ItzLevvie — DISMISSED** [review]

**Review by ItzLevvie — DISMISSED** [review]

**Comment by stephengillie** [comment]
@russellbanks Installation failed in the sandbox due to a hash mismatch.

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  Scope: user
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.multi.win64.installer.msix
+  InstallerSha256: 77873C3244536E309A217C4E0ABFD724EE12AC1C93EE85B8870202869A9E6753
+  SignatureSha256: EA64DC3AB40C58A1C391E09A0145177EE8BF654F1A7F55B0D9D5FF3AB8FF74FC
+  PackageFamilyName: Mozilla.MozillaFirefoxNightly_gmpnhwe7bv608
+- Architecture: x86
+  InstallerType: msix
+  Scope: user
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.multi.win32.installer.msix
+  InstallerSha256: A089692E718A18DC0FB76CB4C68601B8D90745D3EB4767464BB795F095387AA6
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:32]:
```suggestion
  InstallerSha256: 428CCA6FCD15E9B94F003BD7281646EAEE49BFD56621008DE53A1DA13A670E1D
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+- htm
+- html
+- pdf
+- url
+Installers:
+- Architecture: x64
+  InstallerType: msix
+  Scope: user
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.multi.win64.installer.msix
+  InstallerSha256: 77873C3244536E309A217C4E0ABFD724EE12AC1C93EE85B8870202869A9E6753
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:25]:
```suggestion
  InstallerSha256: 186287BEA31B60C582C73B46EF0BC81D0D3AA951AF926B6A722D4B8CDD8FFA11
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  Scope: user
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.multi.win32.installer.msix
+  InstallerSha256: A089692E718A18DC0FB76CB4C68601B8D90745D3EB4767464BB795F095387AA6
+  SignatureSha256: 702F55740294908AAD74064F87AE73E95601F3D172BB91C64F3133E8883BED26
+  PackageFamilyName: Mozilla.MozillaFirefoxNightly_gmpnhwe7bv608
+- Architecture: x64
+  InstallerType: wix
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win64.installer.msi
+  InstallerSha256: 58ECB49C8574C2E647BEA499F79FDCF794E3A94A4AD9C53BD8D11EF0BB34BF53
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:39]:
```suggestion
  InstallerSha256: 2F9315B422510A24D7234DB12F306D9FBE6743EECF17265664DC95C435144C3B
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  InstallerType: wix
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win32.installer.msi
+  InstallerSha256: DCE31F5414882407858276B3996E61E7692BF48AF958C0748F9C2BCC9672B0EC
+  ProductCode: '{1294A4C5-9977-480F-9497-C0EA1E630130}'
+- Architecture: x64
+  InstallerType: exe
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win64.installer.exe
+  InstallerSha256: 76473F6228979A48D8E178097922D9CB7B9623E65DD3E676263F02AE9AF2F12C
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:51]:
```suggestion
  InstallerSha256: F8018D57404165A89DB97ECE078D617EE39142B71D5E3184B9CA7C968AEBF882
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  InstallerSwitches:
+    Silent: /S
+    SilentWithProgress: /S
+  ProductCode: '{1294A4C5-9977-480F-9497-C0EA1E630130}'
+- Architecture: x86
+  InstallerType: exe
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win32.installer.exe
+  InstallerSha256: 3841A94D437E74D98834F8399B90ADE13D8F0BBE1C3A73C309A852CCD2008CBF
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:60]:
```suggestion
  InstallerSha256: B5598A37C5FB3AAC433D4938D348417F6800F4E23FADF9EC050052951B84E1C0
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  InstallerType: wix
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win64.installer.msi
+  InstallerSha256: 58ECB49C8574C2E647BEA499F79FDCF794E3A94A4AD9C53BD8D11EF0BB34BF53
+  ProductCode: '{1294A4C5-9977-480F-9497-C0EA1E630130}'
+- Architecture: x86
+  InstallerType: wix
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win32.installer.msi
+  InstallerSha256: DCE31F5414882407858276B3996E61E7692BF48AF958C0748F9C2BCC9672B0EC
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:45]:
```suggestion
  InstallerSha256: DB2F6C2E10EBACB812FFEB30C8B353260680D237DEAE3452A0DB2E75EE58F45E
```

File: manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  InstallerSha256: 3841A94D437E74D98834F8399B90ADE13D8F0BBE1C3A73C309A852CCD2008CBF
+  InstallerSwitches:
+    Silent: /S
+    SilentWithProgress: /S
+  ProductCode: '{1294A4C5-9977-480F-9497-C0EA1E630130}'
+- Architecture: arm64
+  InstallerType: exe
+  Scope: machine
+  InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-108.0a1.en-US.win64-aarch64.installer.exe
+  InstallerSha256: E9498998C035AA2FD93ECC009677455C14EE5FA290AE635EA7889A0269929437
```
Comment (stephengillie) [inline manifests/m/Mozilla/Firefox/Nightly/108.0a1/Mozilla.Firefox.Nightly.installer.yaml:69]:
```suggestion
  InstallerSha256: E8EC1D10925C81A4456CEAC46E15292E51F2755E316CB2046BF8CEADCD351988
```

**Comment by denelon** [comment]
@russellbanks, we've narrowed down what is happening here. This is a bit of a strange case where the MSIX version of the installer uses a different #.#.#.# syntax than the package version that contains alpha characters.

We don't have a good way to reason about a package with different versions per installer for a single given "version" of a manifest. We've got some ideas, but the fix is quite large.

What would you think about creating two packages?

One for MSIX with well-formed version information, and one for the "Win32" installers with the alpha-numeric versions?

I'd lean in the direction of the "simpler" package ID being the default with the MSIX, and the longer one with the "Win32" identifier containing the other installer types. The golden path seems to be default users to the MSIX version of the installer, and users who specifically want/need the non MSIX installer can still get it with a different package ID.

**Comment by russellbanks** [comment]
Hmm, this is an odd one. I think creating two packages would be the most ideal solution we can get out of this.

If I'm understanding correctly, this should be split into:
`Mozilla.Firefox.Nightly` (MSIX)
&
`Mozilla.Firefox.Nightly.Win32`

I'll create these packages after the weekend.

**Review by ItzLevvie — DISMISSED** [review]

**Review by stephengillie — APPROVED** [review]

**Pull request merged**